src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPoolManager.cs

```diff
@@ -195,16 +201,16 @@ private static HttpConnectionKey GetConnectionKey(HttpRequestMessage request, Ur
                 else
                 {
                     // Tunnel SSL connection through proxy to the destination.
-                    return new HttpConnectionKey(HttpConnectionKind.SslProxyTunnel, isIPv6Address ? uri.Host : uri.IdnHost, uri.Port, sslHostName, proxyUri);
+                    return new HttpConnectionKey(HttpConnectionKind.SslProxyTunnel, isNonLLAIPv6Address ? uri.Host : uri.IdnHost, uri.Port, sslHostName, proxyUri);
                 }
             }
             else if (sslHostName != null)
             {
-                return new HttpConnectionKey(HttpConnectionKind.Https, isIPv6Address ? uri.Host : uri.IdnHost, uri.Port, sslHostName, null);
+                return new HttpConnectionKey(HttpConnectionKind.Https, isNonLLAIPv6Address ? uri.Host : uri.IdnHost, uri.Port, sslHostName, null);
             }
             else
             {
-                return new HttpConnectionKey(HttpConnectionKind.Http, isIPv6Address ? uri.Host : uri.IdnHost, uri.Port, null, null);
+                return new HttpConnectionKey(HttpConnectionKind.Http, isNonLLAIPv6Address ? uri.Host : uri.IdnHost, uri.Port, null, null);
```

Comment:
What about other places we're using IdnHost, like:
https://github.com/caesar1995/corefx/blob/0c08d53b12c4f93e224b447e850746fd1a9b6660/src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnection.cs#L385-L389
Do those need to be updated as well?

Reply:
That's the content in the URL case Tomas has mentioned, I will update them all.